### PR TITLE
Fix calc_perc and add xr kwargs to ensembles

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] HISTORY.rst has been updated

### After merging to master and before closing the branch:
- [ ] bumpversion (minor / major / patch) has been called on `master` branch
- [ ] Tags have been pushed

(The checklist is empty, but I don't think this PR needs anything? It's so small...)

* **What kind of change does this PR introduce?** 
Fixes a bug in `ensembles._calc_perc` where it would assume that the first dimension of the passed dataset was "realization". This fix removes the tests that were made to assure time in the second dimensions (which is useless) and adds a transpose when needed, to bring "realization" as axis 0.

Also, this adds `**xr_kwargs` to  `create_ensemble` so that when datasets are passed as strings or Paths, some options can be given to the opening call. This can be useful to change the default chunking of `{'time': 10}` for mf datasets or pass a `preprocess` callable (which is why I made this change, for my scripts).

* **Does this PR introduce a breaking change?** 
No, only adds options.

### CODE REVIEW

File: tests/test_ensembles.py [hunk 1]
```diff
@@ -112,17 +112,23 @@ def test_create_unequal_times(self):
...
         )
         np.testing.assert_array_equal(
-            np.percentile(ens["tg_mean"][:, 0, 5, 5], 50), out1["tg_mean_p50"][0, 5, 5]
+            np.percentile(ens["tg_mean"].isel(time=0, lon=5, lat=5), 50),
+            out1["tg_mean_p50"].isel(time=0, lon=5, lat=5),
         )
         np.testing.assert_array_equal(
-            np.percentile(ens["tg_mean"][:, 0, 5, 5], 90), out1["tg_mean_p90"][0, 5, 5]
+            np.percentile(ens["tg_mean"].isel(time=0, lon=5, lat=5), 90),
+            out1["tg_mean_p90"].isel(time=0, lon=5, lat=5),
```
Comment (Zeitsperre) [inline tests/test_ensembles.py]: Commas at the end of signatures will fail in Python3.5. Running black with `--target-version py35` will deal with this problem.

File: tests/test_ensembles.py [hunk 2]
```diff
@@ -357,23 +369,27 @@ def test_kmeans_modelweights(self):
         "matplotlib.pyplot" not in sys.modules, reason="matplotlib.pyplot is required"
     )
     def test_kmeans_rsqcutoff_with_graphs(self):
-        ds = xr.open_dataset(self.nc_file)
-
-        # use random state variable to ensure consistent clustering in tests:
-        [ids, cluster, fig_data] = ensembles.kmeans_reduce_ensemble(
-            data=ds.data, method={"rsq_cutoff": 0.9}, random_state=42, make_graph=True
-        )
+        if pytest.importorskip("sklearn", minversion="0.22"):
```
Comment (Zeitsperre) [inline tests/test_ensembles.py]: Not familiar with this feature. Could this be written as a decorator to the test function?

Comment (tlogan2000) [reply tests/test_ensembles.py]: Not sure but we don't need the 'if' apparently @aulemahal tells me

Comment (aulemahal) [reply tests/test_ensembles.py]: I haven't been able to find examples or ways to do this as decorators, but the line could be simply put at the top of the function (no need for `if`). It raises `pytest.Skip` if the package cannot be found with this minimum version.

**Pull request merged**